extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/MicroProfileRestClientEnricher.java

```diff
@@ -184,70 +201,178 @@ public void forSubResourceWebTarget(MethodCreator methodCreator, IndexView index
             ClassInfo subInterfaceClass, MethodInfo rootMethod, MethodInfo subMethod,
             AssignableResultHandle webTarget, BuildProducer<GeneratedClassBuildItem> generatedClasses) {
 
-        Map<String, QueryData> queryParamsByName = new HashMap<>();
-        collectClientQueryParamData(rootInterfaceClass, rootMethod, queryParamsByName);
-        collectClientQueryParamData(subInterfaceClass, subMethod, queryParamsByName);
+        Map<String, ParamData> queryParamsByName = new HashMap<>();
+        collectClientParamData(rootInterfaceClass, rootMethod, queryParamsByName,
+                CLIENT_QUERY_PARAM, CLIENT_QUERY_PARAMS, "ClientQueryParam");
+        collectClientParamData(subInterfaceClass, subMethod, queryParamsByName,
+                CLIENT_QUERY_PARAM, CLIENT_QUERY_PARAMS, "ClientQueryParam");
```

Comment:
Please, move "ClientQueryParam" into a constant and/or provide a custom collectClientParamData method only for query params.